File: files/100-199/185/CH-185.py
import pandas as pd
import re

path = "CH-185 Replace consecutive X.xlsx"
input = pd.read_excel(path,usecols="C:D", skiprows=1, nrows=9)
test = pd.read_excel(path, usecols="F:G", skiprows=1, nrows=9).rename(columns=lambda x: x.split('.')[0])

input['ID'] = [re.sub(r'[Xx]+', 'X', str(x)) for x in input['ID']]

print(input.equals(test))

File: files/100-199/185/CH-185.R
library(tidyverse)
library(readxl)

path = "files/CH-185 Replace consecutive X.xlsx"
input = read_excel(path, range = "C2:D10")
test  = read_excel(path, range = "F2:G10")

result = input %>%
  mutate(ID = str_replace_all(ID, "[Xx]+", "X"))

all.equal(result, test)
#> [1] TRUE